packages/Kokkos/src/DTK_PointCloud_def.hpp

```diff
@@ -0,0 +1,235 @@
+//---------------------------------------------------------------------------//
+/*
+  Copyright (c) 2012, Stuart R. Slattery
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are
+  met:
+
+  *: Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+
+  *: Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  *: Neither the name of the University of Wisconsin - Madison nor the
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \brief DTK_PointCloud_def.hpp
+ * \author Stuart R. Slattery
+ * \brief Point cloud.
+ */
+//---------------------------------------------------------------------------//
+
+#ifndef DTK_POINTCLOUD_DEF_HPP
+#define DTK_POINTCLOUD_DEF_HPP
+
+#include <DTK_DBC.hpp>
+#include <DTK_KokkosHelpers.hpp>
+
+#include <Teuchos_CommHelpers.hpp>
+#include <Teuchos_Ptr.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <limits>
+#include <type_traits>
+
+namespace DataTransferKit
+{
+//---------------------------------------------------------------------------//
+// Local bounding box Kokkos functor.
+//---------------------------------------------------------------------------//
+//! Functor to compute the local bounding box of the point cloud.
+template <class SC, class LO, class GO, class NO>
+class LocalBoundingBoxFunctor
+{
+  public:
+    //! Coordinate type alias.
+    using coordinate_view =
+        typename PointCloud<SC, LO, GO, NO>::coordinate_view;
+    typedef SC value_type[];
```

Comment:
It allows passing everything by value with the same functor interface. See the kokkos user guide for an example of this